mock/py/mockbuild/plugins/package_state.py

```diff
@@ -62,6 +67,55 @@ def _availablePostYumHook(self):
                 self.avail_done = True
                 self.state.finish("Outputting list of available packages")
 
+    @traceLog()
+    def _installedPreBuildHook2(self):
+        filename = "mock-build-environment.json"
+        statename = "Outputting the file " + filename
+        try:
+            with self.buildroot.uid_manager:
+                self.state.start(statename)
+                out_file = os.path.join(self.buildroot.resultdir, filename)
+                chrootpath = self.buildroot.make_chroot_path()
+
+                package_list_cmd  = "rpm --root {0} -qa".format(shlex.quote(chrootpath))
```

Comment:
Adding licenses - they're needed for SBOM generation.

Reply:
It's (a bit) tricky because we need to map the output of `rpm --qf` with the output from `repoquery` (which reorders the stuff according to its own rules, hopefully it is just sorted using NVRA).

The `repoquery --location` can't be combined with `--qf` :-) and there's no `--qf %location` in DNF4's repoquery.

Reply:
Ok, done, but the `signature` field seems undocumented in the Koji Content Generator, and I'm not sure how to get it (therefore there's a hack parsing the %{sigpgp:pgpsig} output for now).  Anyway, we don't use signed packages in Koji/Brew, or?  The current package metadata looks like:
```
            {
                "license": "LicenseRef-Fedora-Public-Domain",
                "name": "filesystem",
                "version": "3.18",
                "release": "23.fc41",
                "arch": "x86_64",
                "epoch": null,
                "nvra": "filesystem-3.18-23.fc41.x86_64",
                "sigmd5": "dc6edb2b7e390e5f0994267d22b9dc1a",
                "signature": null,
                "url": "http://ftp.sh.cvut.cz/fedora/linux/development/rawhide/Everything/x86_64/os/Packages/f/filesystem-3.18-23.fc41.x86_64.rpm"
            },

```

Reply:
We don't use them for building - we use stripped versions of them. But rpms can be imported and/or signed and this info is stored in koji. Also CG-imported content could have been created from such. In case we wouldn't have checksums of the signed content we wouldn't be able to match them for 100%.

Reply:
OK, got an offline confirmation that just the last 8 characters from the `%{sigpgp:pgpsig}` are needed (available/provided e.g. for `rhel-9-x86_64` RPMs from RH's CDN).  PR is updated (thread marked as resolved).

This `rpm -q --qf` + `dnf repoquery` dance is pretty clumsy - @tkopecek proposed to have a dedicated script working with the python3-rpm library - which sounds reasonable, but not trivial in Mock's abstraction layers.  Moving the idea to #1433.